### DISCUSSION TITLE
Power-over-Ethernet option

### DISCUSSION THREAD

**Discussion by wayland** [shanteacontrols/OpenDeck#190] [Ideas]
Hi!  

As an addendum to my suggestion for increased modularity, I'm sure you recall I mentioned the possibility of running it from Power over Ethernet.  

My original idea was to receive PoE, and split that out into:
- A USB port that would go to the OpenDeck MCU (STM Black Pill)
- A USB port that would power a Raspberry Pi 4 (giving me some extra functionality in the box)
- An ethernet port that would go to the Raspberry Pi 4

...but then I had an idea.  So I did a bit of Googling, and found the W5100.  This is a chip that talks TCP/IP on one side, and MCU communication (eg. SPI) on the other side.  
Datasheet: https://www.wiznet.io/wp-content/uploads/wiznethome/Chip/W5100/Document/W5100_DS_V128E.pdf

A little more googling revealed that there's already a MIDI-over-Ethernet protocol: https://en.wikipedia.org/wiki/RTP-MIDI

So, my questions are:
- Do you think that, by the integration of a W5100 that we'd be able to add an Ethernet port to an OpenDeck (ie. are there any pins that might usefully communicate with the W5100, say via SPI)
- Are you aware of anything that would allow us to use RTP-MIDI with OpenDeck?  I found https://github.com/lathoub/Arduino-AppleMIDI-Library but note that the list of supported MCUs is not nearly as large as that for OpenDeck.
- Is this something that interests you, or not really?  

Thoughts?

Thanks!

**Comment by wayland** [comment]
Update: I was thinking a good starting place was:

https://docs.wiznet.io/Product/iEthernet/W5500/w5500-evb-pico#schematic

Links a RP2040 to the W5500 (updated version of the W5100).  

On the RP2040, it uses SPI0 (GPIO 16-19) and 2 other pins (in this example, GPIO 20-21) to communicate.  I realise this clashes with your current allocation for the Traffic/BTLDR pins, and I'm not sure which SPI (if any) you're already using.  

HTH,

**Comment by paradajz** [comment]
> * Is this something that interests you, or not really?

Not really. But I'm also not sure what is your interest here - pure curiosity or would you genuinely find PoE + MIDI over Ethernet useful?

**Reply by wayland** [reply]
I'm expecting to find it useful, but could honestly be wrong about that.  What it gives me is:
- Long cable runs
- Only one cable for power/data (the same as USB)
- The ability to put network monitoring (ping) onto the console; this would be useful for live streaming
- Our other devices (amp, mixer, etc) already communicate via Ethernet; it would make sense to link in to that.  
- Definitely not sure that the MIDI-over-Ethernet idea is going to be useful, except to communicate between control decks if I end up with multiple decks (potentially in different buildings on the same network)

HTH,
